src/nus/cs2103/addressbook/AddressBook.java

```diff
@@ -557,53 +641,60 @@ private static void updateLatestViewedPersonListing(List<String[]> newListing) {
      */
 
     /**
-     * Create storage file if it does not exist. Shows feedback to user.
+     * Creates storage file if it does not exist. Shows feedback to user.
      *
      * @param filePath file to create if not present
      */
-    private static void tryToCreateFileIfMissing(String filePath) {
+    private static void createFileIfMissing(String filePath) {
         final File storageFile = new File(filePath);
+        if (storageFile.exists()) {
+            return;
+        }
+
+        showToUser(String.format(MESSAGE_ERROR_MISSING_STORAGE_FILE, filePath));
+
         try {
-            if (storageFile.createNewFile()) {
-                showToUser(String.format(MESSAGE_ERROR_MISSING_STORAGE_FILE, filePath));
-                showToUser(String.format(MESSAGE_STORAGE_FILE_CREATED, filePath));
-            }
+            storageFile.createNewFile();
+            showToUser(String.format(MESSAGE_STORAGE_FILE_CREATED, filePath));
         } catch (IOException ioe) {
             showToUser(String.format(MESSAGE_ERROR_CREATING_STORAGE_FILE, filePath));
+            exitProgram();
         }
     }
 
     /**
-     * Converts contents of a file into a list of persons. Shows feedback to user.
+     * Converts contents of a file into a list of persons.
+     * Exits program if any errors in reading or decoding was encountered.
      *
      * @param filePath file to load from
-     * @return able to decode: optional containing list of persons in file
-     *         problems decoding: empty optional
+     * @return the list of decoded persons
      */
-    private static Optional<List<String[]>> loadPersonsFromFile(String filePath) {
-        try (final BufferedReader storageReader =
-                     new BufferedReader(new FileReader(filePath))) {
-            final List<String[]> decodedPersons = new ArrayList<>();
+    private static List<String[]> loadPersonsFromFile(String filePath) {
+        final List<String[]> decodedPersons = new ArrayList<>();
+        //TODO: separate file reading from decoding
```

Comment:
In the latest commit, I added these twin methods to the `PERSON METHODS` section:
```java
   /**
     * Encodes list of persons into list of decodable and readable string representations.
     *
     * @param persons to be encoded
     * @return encoded strings
     */
    private static List<String> encodePersonsToStrings(List<String[]> persons) {
        final List<String> encoded = new ArrayList<>();
        for (String[] person : persons) {
            encoded.add(encodePersonToString(person));
        }
        return encoded;
    }

    /**
     * Decode persons from a list of string representations.
     *
     * @param encodedPersons strings to be decoded
     * @return if cannot decode any: empty Optional
     *         else: Optional containing decoded persons
     */
    private static Optional<List<String[]>> decodePersonsFromStrings(List<String> encodedPersons) {
        final List<String[]> decodedPersons = new ArrayList<>();
        for (String encodedPerson : encodedPersons) {
            final Optional<String[]> decodedPerson = decodePersonFromString(encodedPerson);
            if (!decodedPerson.isPresent()) {
                return Optional.empty();
            }
            decodedPersons.add(decodedPerson.get());
        }
        return Optional.of(decodedPersons);
    }
```
Here the decode method does a pure conversion with no side effects; it might be weird for a `convertPersonToLines` method to exit the program if a corrupted line is encountered.